### DISCUSSION TITLE
Efficient variable font design workflow

### DISCUSSION THREAD

**Discussion by bensapirstein** [coldtype/coldtype#134] [Q&A]
I've been using Coldtype to inspect the animation when designing a variable font. I'm working in parallel with a font editing software and I've encountered an issue where every time I finish editing glyphs and want to preview the interpolation I have to restart Coldtype. It seems that the font is being loaded only once, causing this inconvenience.

Is there a way to optimize the workflow and avoid restarting Coldtype each time I want to observe how the edits affected the glyphs interpolation?

**Comment by stenson** [comment]
Great question, I'm assuming you're using `Font.Cacheable` to load the font? You can use just `Font` instead — there's an example of this technique in https://github.com/coldtype/coldtype/blob/main/examples/animations/dswatch.py, which you can work from if you're also using a `.designspace` file to build your variable font — another trick there is to pass the font object to the `watch` kwarg of the `@animation`, that _should_ automatically reload the animation anytime the designspace file changes.

**Comment by bensapirstein** [comment]
Thank you, Rob, much appreciated. 
I've tried the second trick you mentioned, and it works perfectly for reloading the animation. However, when I attempted the first suggestion of using Font instead of Font.Cacheable, I encountered a problem where my stroke-based representations turned into closed contours. Is there a workaround available to address this issue? I remember you mentioning the --keep-overlaps flag, could it be applied in this case? I prefer using this method as it treats the fonts as variable fonts and works seamlessly with their axes.

**Reply by stenson** [reply]
Mm, that makes sense, since the font is being implicitly compiled but there's no way to set that --keep-overlaps flag. A more "manual" option would be to go through and "unclose" the paths by subbing closePath operations for endPath operations, like this:

```python
from coldtype import *

ds = Font(__sibling__("strokefont.designspace"))

def close_to_end(p:P):
    p.explode()
    for contour in p:
        contour.v.value[-1] = ("endPath", ())
    return p.implode()

@animation(tl=60, bg=bw(1))
def broad_nib(f):
    return (StSt("a", ds, 750, nwx0=f.e())
        .mapv(close_to_end)
        .outline(1)
        .align(f.a.r))
```

Any chance that works for you?

**Reply by bensapirstein** [reply]
Thanks, it resolves the closed contour issue, although the alignment doesn't work as effectively as before. Additionally, I noticed a glitch at time 0 when I initially didn't change the value of the parameter passed to outline (which was set to 0.1). I'm curious to understand why this occurred.

```python
from coldtype import *
from coldtype.fx.skia import phototype

fnt = Font("Wasim/Wasim.designspace")

def close_to_end(p:P):
    p.explode()
    for contour in p:
        contour.v.value[-1] = ("endPath", ())
    return p.implode()

@animation(tl=120, bg=bw(1))
def scratch(f):
    return (StSt("ب", fnt, 1000, wght=f.e("eeio"))
            .mapv(close_to_end)
            .outline(1)
            .castshadow(40, f.e("eeio", rng=(60, 100)))
            .align(f.a.r)
            .f(1)
            .ch(phototype(f.a.r, blur=3, cut=90, cutw=30, fill=bw(0)))
            )
```

https://github.com/coldtype/coldtype/assets/14977606/177ebaea-3f46-44ae-8652-5615e83834e5
